pycytominer/operations/noise_removal.py

```diff
@@ -0,0 +1,63 @@
+"""
+Remove noisy features, as defined by features with excessive standard deviation within the same perturbation group.
+"""
+
+import numpy as np
+import pandas as pd
+from pycytominer.cyto_utils import infer_cp_features
+
+
+def noise_removal(
+        population_df, perturb_list, features, samples="all", stdev_cutoff=0.8,
+):
+    """
+
+    Args:
+        population_df: Panda Dataframe which contains all measurement data.
+        perturb_list: The list of unique perturbations corresponding to the rows in population_df. For example,
+            perturb1_well1 and perturb1_well2 would both be "perturb1".
+        features: List of features. Can be inferred or manually supplied.
+        samples: Which rows to use from population_df. Use "all" if applicable.
+        stdev_cutoff: Maximum value for stdev for a given feature to be kept.
+
+    Returns: A list of features to be removed, due to having too high standard deviation within replicate groups.
+
+    """
+    # Subset dataframe
+    if samples != "all":
+        population_df = population_df.loc[samples, :]
+
+    if features == "infer":
+        features = infer_cp_features(population_df)
+
+    population_df = population_df.loc[:, features]
+
+    # Label each row with the identity of its perturbation group
+    population_df['group'] = perturb_list
```

Comment:
Maybe. I was going for allowing the user to set custom perturbation groupings (for example, count each gene as a unique perturb, or each gene-guide combo. I agree that having an inferred default from some kind of metadata column makes sense.

Reply:
I'm about to dig into this review @ruifanp, but I am curious as to how you're thinking about @niranjchandrasekaran 's suggestion here. I like having the option, but I do agree that it's the simplest thing to use an existing metadata column.

Ok, working on my review now! 🤓

Reply:
you can support both options @ruifanp 

```python
if type(perturb) == "str":
    assert perturb in population_df.columns, f"{perturb} not found. Are you sure it is a metadata column?"
    perturb_update = population_df.loc[:, perturb]
else:
   assert len(perturb) == population_df.shape[0], f"The length of input list: {len(perturb)} is not equivalent to your data: {population_df.shape[0]}"
   perturb_update = perturb.copy()

population_df = population_df.assign(group=perturb_update)
```

Note four things:

1. I changed the name of `perturb_list` to a placeholder - if you go this route, which I strongly advise (it's so much easier for the profiling recipe), then you should not indicate that the variable is a list, if it can also be a string (but that's obvious 😄 ) 
2. You will need to reorder when you're subsetting features in the population_df to be after this step
3. If the user inputs a list, it would be smart to assert that the list is the same length as rows in the population_df. This is especially important if the user also sets `samples` to be something other than "all".
4. I did not test this code - it likely needs a fix or two.